### DISCUSSION TITLE
Circuit diagram?

### DISCUSSION THREAD

**Discussion by ustajan** [OpenGammaProject/Open-Gamma-Detector#54] [Q&A] [answered]
Thank you very much for this project!  I am teaching a course on nuclear detection, and I'd like my students to build this detector -- preferably from (near) scratch.  

In particular I am most interested in having them learn about the analog steps between the SiPM output and the microcontroller -- the comparator, the peak detector, and reset (time remaining we will also build the boost converter).  
Can I ask you to share the circuit diagram for the PCB?  I searched for it on the repo but couldn't find it.  
I am most interested in everything "yellow" in the following flow chart:

![Screen Shot 2024-03-14 at 6 28 08 PM](https://github.com/OpenGammaProject/Open-Gamma-Detector/assets/34780293/b2c1cdd5-8322-4b30-b52c-130f39cde3f3)

Much thanks.

**Comment by dc1rdb** [comment, marked as answer]
Hi,
it is available here: https://github.com/OpenGammaProject/Open-Gamma-Detector/blob/main/hardware/schematic.pdf

**Reply by ustajan** [reply]
Thank you!

**Comment by plamenbe** [comment]
Hi there!
Just a note to @ustajan about teaching course. I am also very passioned to teach young people a good stuff. Some years ago I heard about interesting "microcontrollers" called PSoC. They are hybrid devices with user routable analog components inside the chip (OpAmps, comparators, ADCs, DACs, etc.) and MCU (either 8000 family or some ARM variant). Also there are very flexible and robust pathways for interaction between analog and digital domain in the SoC (DMA channels and etc.). Of course these devices are not capable to accomplish demanding scientific tasks solely, but for education I think they are very convenient, because all of analog work could be done by development software for PSoC and in such way to save time and money for discrete components for students (if of course soldering/assembling is not of desire in the course). Also on the Internet there are a couple of projects using PSoC for nuclear electronics. So, may be it is worth to give it a try. 
Just for reference:
 1. [https://en.wikipedia.org/wiki/Cypress_PSoC#:~:text=PSoC%20(programmable%20system%20on%20a,integrated%20analog%20and%20digital%20peripherals](https://en.wikipedia.org/wiki/Cypress_PSoC#:~:text=PSoC%20(programmable%20system%20on%20a,integrated%20analog%20and%20digital%20peripherals)
 2. [www.infineon.com](https://www.infineon.com/cms/en/product/microcontroller/32-bit-psoc-arm-cortex-microcontroller/)
3. [https://www.researchgate.net/publication/327907261_DEVELOPMENT_OF_SINGLE_CHANNEL_ANALYSER_USING_PSoC3_PROCESSOR_MODULE](https://www.researchgate.net/publication/327907261_DEVELOPMENT_OF_SINGLE_CHANNEL_ANALYSER_USING_PSoC3_PROCESSOR_MODULE)
